src/main/groovy/com/google/protobuf/gradle/ProtobufExtension.groovy

```diff
@@ -80,13 +79,21 @@ abstract class ProtobufExtension {
   }
 
   String getGeneratedFilesBaseDir() {
-    return generatedFilesBaseDir
+    return generatedFilesBaseDirProperty.get()
   }
 
+  @Deprecated
   void setGeneratedFilesBaseDir(String generatedFilesBaseDir) {
-    this.generatedFilesBaseDir = generatedFilesBaseDir
+    generatedFilesBaseDirProperty.set(generatedFilesBaseDir)
   }
 
+  /**
+   * The base directory of generated files. The default is
+   * "${project.buildDir}/generated/source/proto".
+   */
+  @PackageScope
```

Comment:
Add `@Input` ? Looks it should be public.[ Example in gradle docs](https://docs.gradle.org/current/userguide/lazy_configuration.html#connecting_properties_together)

Reply:
`@Input` is for tasks and this isn't a task. I don't see any need for it to be public.

Reply:
Ouch, my bad.